tracer/src/Datadog.Trace/Headers/Ip/RequestIpExtractor.cs

```diff
@@ -0,0 +1,71 @@
+// <copyright file="RequestIpExtractor.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.Headers.Ip
+{
+    internal static class RequestIpExtractor
+    {
+        private static readonly IReadOnlyList<string> IpHeaders = new[] { "x-forwarded-for", "x-real-ip", "client-ip", "x-forwarded", "x-cluster-client-ip", "forwarded-for", "forwarded", "via", "true-client-ip" };
+        internal static readonly IReadOnlyList<string> MainHeaders = new[] { "user-agent", "referer" };
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(RequestIpExtractor));
+
+        internal static IpInfo ExtractIpAndPort(Func<string, string> getHeader, string customIpHeader, bool isSecureConnection, IpInfo peerIpFallback)
+        {
+            IpInfo result = null;
+            if (!string.IsNullOrEmpty(customIpHeader))
+            {
+                var value = getHeader(customIpHeader);
+                if (!string.IsNullOrEmpty(value))
+                {
+                    result = IpExtractor.RealIpFromValue(value, isSecureConnection);
+                    if (result == null)
+                    {
+                        Log.Warning("A custom header for ip with value {value} was configured but no ip could be read", value);
+                        return null;
+                    }
+                }
+            }
+
+            string potentialIp = null;
+            foreach (var headerIp in IpHeaders)
+            {
+                var headerValue = getHeader(headerIp);
+                if (!string.IsNullOrEmpty(headerValue))
+                {
+                    if (!string.IsNullOrEmpty(potentialIp) || !string.IsNullOrEmpty(customIpHeader))
+                    {
+                        Log.Warning("Multiple ip headers have been found, none will be reported", potentialIp);
+                        return null;
+                    }
+
+                    potentialIp = headerValue;
+                }
+            }
+
+            return result ?? (!string.IsNullOrEmpty(potentialIp) ? IpExtractor.RealIpFromValue(potentialIp, isSecureConnection) : peerIpFallback);
```

Comment:
if you accept the other suggestion above.

```suggestion
            return (!string.IsNullOrEmpty(potentialIp) ? IpExtractor.RealIpFromValue(potentialIp, isSecureConnection) : peerIpFallback);
```